lib/deflate.rb

```diff
@@ -0,0 +1,23 @@
+class Array
+  def deflate(*args, &block)
+    original = dup
```

Comment:
Thanks for feedback Sam! :)

Yeah, It's so that when i call #shift in the below two methods, the actual array doesn't lost it's first item. I tried to avoid it, but seemed the cleanest way of dealing with the issue?